lenskit-core/src/main/java/org/lenskit/rerank/Rescorer.java

```diff
@@ -0,0 +1,28 @@
+package org.lenskit.rerank;
+
+import org.lenskit.api.Result;
+import org.lenskit.api.ResultList;
+
+import java.util.List;
+
+/**
+ * Interface for classes that produce scores for the value of adding an item to a list of recommended items. This class
+ * is used in {@link GreedyRerankingItemRecommender} to define the metric to optimize when recommending items.
+ *
+ * While this interface does not have support for hard filtering, a filter behavior can be approximated by returning
+ * {@link Double#MIN_VALUE}. Filtered items may still be returned, but only after all non-filtered items are eliminated.
+ *
+ * @author Daniel Kluver
+ */
+public interface Rescorer {
```

Comment:
Sure. Options (feel free to remix until we find truth):
 * `ItemSetScorer`
 * `RecommendationCandidateEvaluator`
 * `GreedyRerankingCriteria`